lib/search-bar.jsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import appState from './flux/app-state';
+import { tracks } from './analytics';
+import NewNoteIcon from './icons/new-note';
+import SearchField from './search-field';
+import TagsIcon from './icons/tags';
+
+const {
+	newNote,
+	toggleNavigation,
+} = appState.actionCreators;
+const { recordEvent } = tracks;
+
+export const SearchBar = ( {
+	onNewNote,
+	onToggleNavigation,
+	showTrash,
+} ) =>
+	<div className="search-bar theme-color-border">
+		<button className="button button-borderless" onClick={ onToggleNavigation } title="Tags">
+			<TagsIcon />
+		</button>
+		<SearchField />
+		<button
+			className="button button-borderless"
+			disabled={ showTrash }
+			onClick={ onNewNote }
+			title="New Note"
+		>
+			<NewNoteIcon />
+		</button>
+	</div>;
```

Comment:
just a personal preference, and maybe I've broken my own preference, but what do you think about surrounding this in parens just to visually delineate the JSX?

```js
} ) => (
	<div…
	</div>
);
```